Lab2.ipynb

```diff
@@ -0,0 +1,636 @@
+      "text/plain": [
+       "<Figure size 1080x360 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%matplotlib inline\n",
+    "import numpy as np\n",
+    "import matplotlib\n",
+    "import matplotlib.pyplot as plt\n",
+    "import scipy\n",
+    "from scipy import stats\n",
+    "\n",
+    "x0 = np.arange(0, 15)\n",
+    "dist = stats.poisson.pmf(x0, 5)\n",
+    "\n",
+    "conv1 = np.convolve(dist, dist)\n",
+    "x1 = np.linspace(0, len(conv1), len(conv1))\n",
+    "\n",
+    "conv2 = np.convolve(conv1, conv1)\n",
+    "x2 = np.linspace(0, len(conv2), len(conv2))\n",
+    "\n",
+    "conv3 = np.convolve(conv2, conv2)\n",
+    "x3 = np.linspace(0, len(conv3), len(conv3))\n",
+    "\n",
+    "conv4 = np.convolve(conv3, conv3)\n",
+    "x4 = np.linspace(0, len(conv4), len(conv4))\n",
+    "\n",
+    "fig, ax = plt.subplots(1, 1, figsize=(15, 5))\n",
+    "\n",
+    "ax.step(x0, dist, label='Probability Distribution')\n",
+    "ax.step(x1, conv1, label='First Convolution')\n",
+    "ax.step(x2, conv2, label='Second Convolution')\n",
+    "ax.step(x3, conv3, label='Third Convolution')\n",
+    "ax.step(x4, conv4, label='Fourth Convolution')\n",
```

Comment:
I think technically it's not accurate to use 1,2,3,4th convolution here, since you're convoluting already convoluted functions (e.g. conv2 = dist * dist * dist * dist) and so the convolution count goes up by the square of the iteration (I believe).